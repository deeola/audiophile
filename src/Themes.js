export const Colors = {
  orange: "#D87D4A",
  lightOrange: "#fbaf85",
  white: "#FFFFFF",
  lightBlack: "#101010",
  black: "#000000",
  gray: "#F1F1F1",
  lightgray: "#FAFAFA",
};
