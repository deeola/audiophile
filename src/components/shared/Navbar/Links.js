import React from 'react'
import { List } from './Navbarstyling'
import PropTypes from 'prop-types';

const Links = ({label, to }) => {
  return (
    <List to={to} data-testid={`list-${label}`} >{label}</List>
  )
}
Links.propTypes = {
  primary: PropTypes.bool,
  label: PropTypes.string.isRequired,
};

export default Links