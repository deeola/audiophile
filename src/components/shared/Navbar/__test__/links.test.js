import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import Links from "../Links";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import { Colors } from "../../../../Themes";

afterEach(cleanup);

const theme = {
  colors: Colors,
};



it("renders navbar correctly", () => {
  const div = document.createElement("a");
  ReactDOM.render(
    <BrowserRouter>
      <Links></Links>
    </BrowserRouter>,
    div
  );
});

// it("renders Home Link correctly", () => {
//   const { getByTestId } = render(
//     <BrowserRouter>
//       <Links to={"/"} label={"Home"}></Links>
//     </BrowserRouter>
//   );

//   expect(getByTestId("list")).toHaveTextContent("Home");
// });

// it("renders Headphone  Link correctly", () => {
//   const { getByTestId } = render(
//     <BrowserRouter>
//       <Links to={"/Headphones"} label={"Headphones"}></Links>
//     </BrowserRouter>
//   );
//   expect(getByTestId("list")).toHaveTextContent("Headphones");
// });

// it("renders Speakers Link correctly", () => {
//   const { getByTestId } = render(
//     <BrowserRouter>
//       <Links to={"/Speakers"} label={"Speakers"}></Links>
//     </BrowserRouter>
//   );
//   expect(getByTestId("list")).toHaveTextContent("Speakers");
// });

// it("renders Earphones Link correctly", () => {
//   const { getByTestId } = render(
//     <ThemeProvider theme={theme}>
//       <BrowserRouter>
//         <Links to={"/Earphones"} label={"Earphones"}></Links>
//       </BrowserRouter>
//     </ThemeProvider>
//   );
//   expect(getByTestId("list")).toHaveTextContent("Earphones");
// });

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Links to={"/"} label={"Home"}></Links>
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
