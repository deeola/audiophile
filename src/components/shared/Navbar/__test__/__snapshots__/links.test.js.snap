// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<a
  className="sc-gKXOVf cBwxmB"
  data-testid="list-Home"
  href="/"
  onClick={[Function]}
>
  Home
</a>
`;
